projects/plugins/boost/app/assets/src/js/features/page-cache/health/health.tsx

```diff
@@ -1,48 +1,82 @@
+	PageCacheRootDirNotWritableNotice,
+	UnableToWriteToAdvancedCacheNotice,
+	WPCacheDefinedNotTrueNotice,
+	WPConfigNotWritableNotice,
+	WPContentNotWritableNotice,
+} from './error-notices';
+import SwitchToBoost from '../switch-to-boost/switch-to-boost';
+
 type HealthProps = {
 	error?: PageCacheError;
-	setup?: ReturnType< typeof usePageCacheSetup >;
+	setup: ReturnType< typeof usePageCacheSetup >;
 	setError: ( error: PageCacheError ) => void;
 };
 
 const Health = ( { setup, error, setError }: HealthProps ) => {
-	const [ , setPageCache ] = useSingleModuleState( 'page_cache' );
+	const [ , setModuleState ] = useSingleModuleState( 'page_cache', state => {
+		// Run setup if the module is being enabled
+		if ( state ) {
+			setup.mutate();
+		}
+	} );
 	// Was there a problem trying to setup cache?
-	const errorData = getErrorData( error );
-	const [ canReset, setCanReset ] = useState( false );
+	const [ doingRevert, setDoingRevert ] = useState( false );
 
 	useEffect( () => {
 		if ( setup?.isError && error && ! error.dismissed ) {
-			setCanReset( true );
+			setDoingRevert( true );
 		}
-	}, [ setup?.isError, error, setCanReset ] );
+	}, [ setup?.isError, error, setDoingRevert ] );
 
 	useEffect( () => {
-		if ( canReset ) {
-			setCanReset( false );
-			setPageCache( false );
+		if ( doingRevert ) {
+			setModuleState( false );
+			setDoingRevert( false );
 		}
-	}, [ canReset, setCanReset, setPageCache ] );
-
-	return (
-		<>
-			{ errorData && error && error.dismissed !== true && (
-				<Notice
-					level="warning"
-					title={ errorData.title }
-					onClose={ () => {
-						setError( { ...error, dismissed: true } );
-					} }
-				>
-					<p>{ errorData.message }</p>
-				</Notice>
-			) }
-		</>
-	);
+	}, [ doingRevert, setDoingRevert, setModuleState ] );
+
+	if ( ! error || error.dismissed ) {
+		return null;
+	}
+
+	const onClose = () => {
+		setError( { ...error, dismissed: true } );
+	};
+
+	const code = typeof error === 'string' ? error : error.code;
+	const notices: { [ key: string ]: ReactNode } = {
+		'failed-settings-write': <FailedSettingsWriteNotice onClose={ onClose } />,
+		'wp-content-not-writable': <WPContentNotWritableNotice onClose={ onClose } />,
+		'not-using-permalinks': <NotUsingPermalinksNotice onClose={ onClose } />,
+		'advanced-cache-incompatible': <AdvancedCacheIncompatibleNotice onClose={ onClose } />,
+		'advanced-cache-for-super-cache': (
+			<AdvancedCacheForSuperCacheNotice
+				actions={ [
+					<SwitchToBoost key={ 'switch-to-boost' } onSwitch={ () => setModuleState( true ) } />,
+				] }
+				onClose={ onClose }
+			/>
+		),
+		'unable-to-write-to-advanced-cache': <UnableToWriteToAdvancedCacheNotice onClose={ onClose } />,
+		'wp-cache-defined-not-true': <WPCacheDefinedNotTrueNotice onClose={ onClose } />,
+		'page-cache-root-dir-not-writable': <PageCacheRootDirNotWritableNotice onClose={ onClose } />,
+		'wp-config-not-writable': <WPConfigNotWritableNotice onClose={ onClose } />,
+	};
```

Comment:
Minor notpick - this instantiates all notices even if they'll remain unused. Moving them to a function would be a micro-optimization though.

Reply:
We moved away from that micro-optimization in this PR for clarity

Reply:
I am unsure what you mean. Feel free to commit an update. Or, give me an example.